> 
> 
> See some in-line comments. The global_sum and its new test looks good.
> 
> I still have some problems with the compiler. When compiling without MPI, using just gfortran but with the -fopenmp flag I am getting the following error (note this is needed for the `manual_verison/psykal_omp` of nemolite2d):
> 
> ```
> gfortran -Wall -Wsurprising -Wuninitialized -faggressive-function-elimination -Ofast -mtune=native -finline-limit=50000 -fopt-info-all=gnu_opt_report.txt -march=core2 -mtune=core2 -ffree-line-length-none  -fopenmp -c /home/sergi/workspace/euroexa/PSycloneBench/shared/dl_esm_inf/finite_difference/src//grid_mod.f90
> /home/sergi/workspace/euroexa/PSycloneBench/shared/dl_esm_inf/finite_difference/src//grid_mod.f90:353:0:
> 
>            do ji = xstart-1, xstop+1
> 
> Error: ‘xstart’ not specified in enclosing ‘parallel’
> /home/sergi/workspace/euroexa/PSycloneBench/shared/dl_esm_inf/finite_difference/src//grid_mod.f90:353:0: Error: enclosing ‘parallel’
> /home/sergi/workspace/euroexa/PSycloneBench/shared/dl_esm_inf/finite_difference/src//grid_mod.f90:353:0: Error: ‘xstop’ not specified in enclosing ‘parallel’
> /home/sergi/workspace/euroexa/PSycloneBench/shared/dl_esm_inf/finite_difference/src//grid_mod.f90:353:0: Error: enclosing ‘parallel’
> ```
> 
> This again can be solved by removing the default(none) clause or setting the shared() variables. but I am a bit puzzled as this was already on master and I didn't had this issue before.
> 
I too don't get this problem but then I see from the Makefile that we don't build dl_esm_inf with OpenMP enabled - just the mini-app itself. Therefore we will only build dl_esm_inf with OpenMP if we set F90FLAGS to include "-fopenmp" before we make. Given that we don't have explicit support in dl_esm_inf for building with OpenMP I think we should leave this for the moment.

--- a/finite_difference/src/grid_mod.f90
+++ b/finite_difference/src/grid_mod.f90
@@ -348,7 +348,7 @@ contains
 !> TODO should use thread tiling here but that is currently only set-up
 !! inside a field object.
 !$OMP PARALLEL DO schedule(runtime), default(none), private(ji,jj), &
-!$OMP shared(grid, tmask)
+!$OMP shared(grid, tmask, ystart, ystop, xstart, xstop)
        do jj = ystart-1, ystop+1
           do ji = xstart-1, xstop+1
              ! Copy in values

--- a/finite_difference/tests/dist_mem/Makefile
+++ b/finite_difference/tests/dist_mem/Makefile
@@ -33,11 +33,15 @@
 # environment variables. e.g. to use gfortran:
 # 
 # export F90=mpif90
+# export MPIRUN=mpirun
 # export F90FLAGS="-O3"
 # export AR=ar
 # make
 #
 all: tests
+
+# How we launch MPI jobs
+MPIRUN ?= mpirun
 
 # Location of the dl_esm_inf library
 INF_DIR=../..
@@ -46,24 +50,22 @@ INF_LIB = ${INF_DIR}/src/lib_fd.a
 
 .PHONY: test_xsplit test_ysplit test_xysplit inf_lib
 
-# Compile and run tests. This is a very basic target and assumes that
-# OpenMPI's 'mpirun' is available (because of the way we set environment
-# variables).
+# Compile and run tests. This is a very basic target.
 tests: test_xsplit test_ysplit test_xysplit test_gsum
 
 test_xsplit: inf_lib test_halos.exe
-	mpirun -x JPIGLO=10 -x JPJGLO=4 -np 2 ./test_halos.exe
+	JPIGLO=10 JPJGLO=4 ${MPIRUN} -np 2 ./test_halos.exe
 
 test_ysplit: inf_lib test_halos.exe
-	mpirun -x JPIGLO=4 -x JPJGLO=10 -np 2 ./test_halos.exe
+	JPIGLO=4 JPJGLO=10 ${MPIRUN} -np 2 ./test_halos.exe
 
 test_xysplit: inf_lib test_halos.exe
-	mpirun -x JPIGLO=10 -x JPJGLO=10 -np 4 ./test_halos.exe
-	mpirun -x JPIGLO=10 -x JPJGLO=10 -np 6 ./test_halos.exe
+	JPIGLO=10 JPJGLO=10 ${MPIRUN} -np 4 ./test_halos.exe
+	JPIGLO=10 JPJGLO=10 ${MPIRUN} -np 6 ./test_halos.exe
 
 test_gsum: inf_lib test_gsum.exe
-	mpirun -x JPIGLO=4 -x JPJGLO=10 -np 4 ./test_gsum.exe
-	mpirun -x JPIGLO=4 -x JPJGLO=10 -np 6 ./test_gsum.exe
+	JPIGLO=4 JPJGLO=10 ${MPIRUN} -np 4 ./test_gsum.exe
+	JPIGLO=4 JPJGLO=10 ${MPIRUN} -np 6 ./test_gsum.exe
 
 inf_lib:
 	${MAKE} -C ${INF_DIR} F90=${F90} dm_fd_lib

